services/web/client/source/class/qxapp/component/widget/FileManager.js

```diff
@@ -0,0 +1,319 @@
+qx.Class.define("qxapp.component.widget.FileManager", {
+  extend: qx.ui.core.Widget,
+
+  construct: function(nodeModel) {
+    this.base(arguments);
+
+    this.set({
+      nodeModel: nodeModel
+    });
+
+    let fileManagerLayout = new qx.ui.layout.VBox(10);
+    this._setLayout(fileManagerLayout);
+
+    let treesLayout = this._createChildControlImpl("treesLayout");
+
+    let nodeTree = this.__nodeTree = this._createChildControlImpl("nodeTree");
+    treesLayout.add(nodeTree, {
+      flex: 1
+    });
+    nodeTree.addListener("changeSelection", this.__itemSelected, this);
+
+    let userTree = this.__userTree = this._createChildControlImpl("userTree");
+    treesLayout.add(userTree, {
+      flex: 1
+    });
+    userTree.addListener("changeSelection", this.__itemSelected, this);
+
+    let selectedFileLayout = this._createChildControlImpl("selectedFileLayout");
+    {
+      let selectedLabel = this.__selectedLabel = new qx.ui.basic.Label().set({
+        decorator: "main",
+        backgroundColor: "white",
+        minWidth: 300,
+        height: 24
+      });
+
+      let downloadBtn = new qx.ui.form.Button().set({
+        icon: "@FontAwesome5Solid/cloud-download-alt/24"
+      });
+      downloadBtn.addListener("execute", function(e) {
+        this.__downloadFile();
+      }, this);
+
+      let deleteBtn = new qx.ui.form.Button().set({
+        icon: "@FontAwesome5Solid/trash-alt/24"
+      });
+      deleteBtn.addListener("execute", function(e) {
+        this.__deleteFile();
+      }, this);
+
+      selectedFileLayout.add(selectedLabel, {
+        flex: 1
+      });
+      selectedFileLayout.add(downloadBtn);
+      selectedFileLayout.add(deleteBtn);
+    }
+
+    let closeBtn = this.__closeBtn = this._createChildControlImpl("closeButton");
+    closeBtn.setEnabled(false);
+    closeBtn.addListener("execute", function() {
+      console.log("close");
+    }, this);
+
+    this.__reloadNodeTree();
+    this.__reloadUserTree();
+  },
+
+  properties: {
+    nodeModel: {
+      check: "qxapp.data.model.NodeModel"
+    }
+  },
+
+  members: {
+    __nodeTree: null,
+    __userTree: null,
+    __currentUserId: "ODEI-UUID",
+    __selectedLabel: null,
+    __selection: null,
+    __closeBtn: null,
+
+    _createChildControlImpl: function(id) {
+      let control;
+      switch (id) {
+        case "treesLayout":
+          control = new qx.ui.container.Composite(new qx.ui.layout.HBox(10));
+          this._add(control, {
+            flex: 1
+          });
+          break;
+        case "nodeTree":
+        case "userTree":
+          control = new qx.ui.tree.Tree();
```

Comment:
maybe use `tree.VirtualTree` everywhere

Reply:
Agree. That would mean changing how the model is built. We could create a list of improvements to tackle once we have everything working.